atomic_reactor/plugins/pre_fetch_sources.py

```diff
@@ -0,0 +1,208 @@
+"""
+Copyright (c) 2019 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import absolute_import
+
+import os
+import tempfile
+
+import requests
+from six.moves.urllib.parse import urlparse
+
+from atomic_reactor.constants import PLUGIN_FETCH_SOURCES_KEY, DEFAULT_DOWNLOAD_BLOCK_SIZE
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.plugins.pre_reactor_config import (
+    get_koji,
+    get_koji_path_info,
+    get_koji_session,
+    get_config
+)
+from atomic_reactor.util import get_retrying_requests_session
+
+
+class FetchSourcesPlugin(PreBuildPlugin):
+    """Download sources that may be used in further steps to compose Source Containers"""
+    key = PLUGIN_FETCH_SOURCES_KEY
+    is_allowed_to_fail = False
+    DOWNLOAD_DIR = 'image_sources'
+
+    def __init__(
+        self, tasker, workflow, koji_build_id=None, koji_build_nvr=None, signing_intent=None,
+        kojihub=None, kojiroot=None, koji_proxy_user=None, koji_ssl_certs=None, koji_principal='',
+        koji_keytab=''
```

Comment:
Maybe we should just take the koji information from the reactor config. Is there any benefit in allowing the plugin to override them?

Reply:
I allowed overriding koji info to be consistent with other plugins getting a koji session. TBH I do not see much benefit on allowing it. @rcerven do you know of any any currently used use case for this in other plugins?

Reply:
I think they just do that for compatibility, since that's how it was done before the reactor config provided it.

Reply:
see e.g. #1280 which uses Koji but just gets config with no fallback.

Reply:
it was just for backwards compatibility during transition to reactor_config_map

for new plugins we should use just reactor_config_map

Reply:
removed koji config overriding capabilities